rasa/nlu/extractors/mitie_entity_extractor.py

```diff
@@ -133,50 +187,105 @@ def _prepare_mitie_sample(training_example: Message) -> Any:
                 continue
         return sample
 
-    def process(self, message: Message, **kwargs: Any) -> None:
-        mitie_feature_extractor = kwargs.get("mitie_feature_extractor")
-        if not mitie_feature_extractor:
-            raise Exception(
-                "Failed to train 'MitieFeaturizer'. "
-                "Missing a proper MITIE feature extractor."
+    def process(
+        self, messages: List[Message], mitie_model: MitieModel,
+    ) -> List[Message]:
+        """Extracts entities from messages and appends them to the attribute.
+
+        If no patterns where found during training, then the given messages will not
+        be modified. In particular, if no `ENTITIES` attribute exists yet, then
+        it will *not* be created.
+
+        If no pattern can be found in the given message, then no entities will be
+        added to any existing list of entities. However, if no `ENTITIES` attribute
+        exists yet, then an `ENTITIES` attribute will be created.
+
+        Returns:
+           the given list of messages that have been modified
+        """
+        if not self._ner:
+            return messages
+
+        for message in messages:
+            entities = self._extract_entities(message, mitie_model=mitie_model)
+            extracted = self.add_extractor_name(entities)
+            message.set(
+                ENTITIES, message.get(ENTITIES, []) + extracted, add_to_output=True
             )
+        return messages
+
+    def _extract_entities(
+        self, message: Message, mitie_model: MitieModel,
+    ) -> List[Dict[Text, Any]]:
+        """Extract entities of the given type from the given user message.
+
+        Args:
+            message: a user message
+            mitie_model: MitieModel containing a `mitie.total_word_feature_extractor`
 
-        ents = self.extract_entities(
-            message.get(TEXT), message.get(TOKENS_NAMES[TEXT]), mitie_feature_extractor
+        Returns:
+            a list of dictionaries describing the entities
+        """
+        text = message.get(TEXT)
+        tokens = message.get(TOKENS_NAMES[TEXT])
+
+        entities = []
+        token_texts = [token.text for token in tokens]
+        mitie_entities = self._ner.extract_entities(
+            token_texts, mitie_model.word_feature_extractor,
         )
-        extracted = self.add_extractor_name(ents)
-        message.set(ENTITIES, message.get(ENTITIES, []) + extracted, add_to_output=True)
+        for e in mitie_entities:
```

Comment:
Where is this new code coming from?

Reply:
this isn't new - it's `extract_entities` in the old code (without underscore)

Reply:
oh ok I see, the re-ordering of the methods makes it hard to see what has changed.

Reply:
oh yeah sorry about that - the extract is not public anymore so I thought it was strange to have that first and before the place where it's used